src/server/lib/okta/api/authentication.ts

```diff
@@ -129,6 +132,14 @@ export const resetPassword = async (body: {
   newPassword: string;
 }): Promise<AuthenticationTransaction> => {
   const path = buildUrl('/api/v1/authn/credentials/reset_password');
+
+  if (await isBreachedPassword(body.newPassword)) {
```

Comment:
I wonder if this should be so tightly coupled with our Okta API method here. If I called `resetPassword` I think i'd expect it to just send my request directly to Okta and not apply any extra checks.

I realise that in most, if not all of the cases we use this method we want to apply this check, but the pattern across our codebase appears to be that these methods are purely calls to the related API endpoint. Perhaps a version of this called `resetPasswordWithBreachedCheck`, or just a boolean argument to this method could be more explicit?

Reply:
Yeah I see what you're getting at, but in this case I think we want a tight coupling to the method, as in all cases where we update a password in Okta we should do a breached check first for our security compliance.

By separating them out into two different methods, future developers might accidently use the normal `resetPassword` method rather than the `resetPasswordWithBreachedCheck` method without realising.

I'll update the documentation so that this reasoning is clear.